collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -290,6 +291,29 @@ public void exit() {
         formSessionRepository.clear(sessionId);
     }
 
+    public void validate() {
+        isLoading.setValue(true);
+        scheduler.immediate(
+                () -> {
+                    Integer result = null;
+                    try {
+                        result = formController.validateAnswers(true);
+                    } catch (JavaRosaException e) {
+                        error.setValue(new NonFatal(e.getMessage()));
+                    }
+
+                    return result;
+                }, result -> {
+                    isLoading.setValue(false);
+
+                    if (result != null && result != FormEntryController.ANSWER_OK) {
+                        refresh();
+                        failedConstraint.setValue(new FailedConstraint(formController.getFormIndex(), result));
```

Comment:
This is not going to work with field-list since the index will be the index of the group not of the particular question. As a result, it will crash the app. I will take over since @seadowg is off.